src/client.c

```diff
@@ -548,6 +548,107 @@ bool handle_auth_query_response(PgSocket *client, PktHdr *pkt) {
 	return true;
 }
 
+/*
+ * skip_ws returns a pointer to the first non whitespace character
+ * in the given string
+ */
+static inline const char *skip_ws(const char *position)
+{
+	while (*position && isspace(*position))
+		position++;
+	return position;
+}
+
+/*
+ * read_escaped_token reads a token that might be escaped using backslashes
+ * from the escaped_string_ptr. The token is written in unescaped form to the
+ * unescaped_token buffer. escape_string_ptr is set to the character right
+ * after the token.
+ */
+_MUSTCHECK
+static bool read_escaped_token(const char **escaped_string_ptr, struct MBuf *unescaped_token)
+{
+	const char *position = *escaped_string_ptr;
+	const char *unwritten_start = position;
+	while (*position)
+	{
+		if (*position == '\\') {
+			if (!mbuf_write(unescaped_token, unwritten_start, position-unwritten_start))
+				return false;
+			position++;
+			unwritten_start = position;
+			if (!*position)
+				break;
+		} else if (isspace(*position)) {
+			break;
+		}
+		position++;
+	}
+	if (!mbuf_write(unescaped_token, unwritten_start, position-unwritten_start))
+		return false;
+	if (!mbuf_write_byte(unescaped_token, '\0'))
+		return false;
+	*escaped_string_ptr = position;
+	return true;
+}
+
+/*
+ * set_startup_options takes the value of to the "options" startup parameter
+ * and uses it to set the parameters that are embeded in this value.
+ *
+ * It only supports the following type of postgres command line argument:
+ * -c config=value
+ *
+ * The reason that we don't support all arguments is to keep the parsing simple
+ * an this is by far the argument that's most commonly used in practice in the
+ * options startup paramater. Also all other postgres command line arguments
+ * can be rewritten to this form:
+ */
+static bool set_startup_options(PgSocket *client, const char *options)
+{
+	char arg_buf[400];
+	struct MBuf arg;
+	const char *position = options;
+	mbuf_init_fixed_writer(&arg, arg_buf, sizeof(arg_buf));
+	slog_debug(client, "received options: %s", options);
+	while (*position) {
+		const char *key_string, *value_string;
+		char *equals;
+		mbuf_rewind_writer(&arg);
+		position = skip_ws(position);
+		if (strncmp("-c", position, 2) != 0)
+			goto fail;
+		position += 2;
+		position = skip_ws(position);
+
+		if (!read_escaped_token(&position, &arg)) {
+			disconnect_client(client, true, "unsupported options startup parameter: parameter too long");
+			return false;
+		}
+
+		equals = strchr((char *) arg.data, '=');
+		if (!equals)
+			goto fail;
+		*equals = '\0';
```

Comment:
It is a matter of style but I will suggest anyway.

`*equals++ = '\0';`

and remove `+ 1` 2 lines below.

Reply:
I'll leave this as is, I think it's clearer this way. Mostly because the variable is called `equals`, thus logically pointing to the equals character. If it was the `position` variable I'd have agreed with the suggestion.